website/docs/d/generic_secret_list.html.md

```diff
@@ -0,0 +1,71 @@
+---
+layout: "vault"
+page_title: "Vault: vault_generic_secret_list data source"
+sidebar_current: "docs-vault-datasource-generic-secret_list"
+description: |-
+  Lists all secrets under a given KV-V1 path
+---
+
+# vault\_generic\_secret
+
+Lists all secrets under a given KV-V1 path in Vault.
+
+This resource is primarily intended to be used with
+[Vault's "generic" secret backend](https://www.vaultproject.io/docs/secrets/generic/index.html),
+but it is also compatible with any other Vault endpoint that supports
+the `vault read` command.
+
+~> **Important** All data retrieved from Vault will be
+written in cleartext to state file generated by Terraform, will appear in
+the console output when Terraform runs, and may be included in plan files
+if secrets are interpolated into any resource attributes.
+Protect these artifacts accordingly. See
+[the main provider documentation](../index.html)
+for more details.
+
+## Example Usage
+
+### Generic secret List
+
+```hcl
+resource "vault_mount" "kv" {
+  path        = "kv"
+  type        = "kv"
+  description = "KV Version 1 secret engine mount"
+  options     = { 
+    version = "1"
+  }
+}
+
+resource "vault_generic_secret" "secret" {
+  path = "${vault_mount.kv.path}/baz"
+
+  data_json = <<EOT
```

Comment:
Suggest using `jsonencode()` instead of the HERE doc syntax.

Reply:
Updated in latest commits